### PR TITLE
fix redis + where + pageError

### PR DESCRIPTION
**Pull request by jinmarcus**
fix redis + where + pageError

### CODE REVIEW

File: system/Database/BaseBuilder.php [hunk 1]
```diff
@@ -574,8 +574,13 @@ public function join($table, $cond, $type = '', $escape = null)
 	 *
 	 * @return    BaseBuilder
 	 */
-	public function where($key, $value = null, $escape = null)
+	public function where($key = null, $value = null, $escape = null)
```
Comment (lonnieezell) [inline system/Database/BaseBuilder.php]: Sorry. Can't accept this one. `$key` is always required and it should fail loudly when the developer forgets to add it.

Comment (jinmarcus) [reply system/Database/BaseBuilder.php]: ok,Only follow this rule

**Comment by lonnieezell** [comment]
I wasn't thinking on your previous mention about where's and null. I don't think it's right to allow that in this case as it could be a bit dangerous and hard to track down during debugging. Sorry about that.

**Comment by lonnieezell** [comment]
@jinmarcus Are you going to be able to revise this PR by removing the one fix so I can merge the others?

**Comment by jim-parry** [comment]
Being picky, but we expect commits to be GPG-signed :)

**Pull request closed without merging**